game/utils/map.py

```diff
@@ -0,0 +1,148 @@
+
+        self.map = blank_map
+
+    def __str__(self) -> str:
+        """
+        Converts the map to a string.
+        :return: Map as a string.
+        """
+        map_to_str = ""
+        for row in self.map:
+            map_to_str += "".join(["#" if cell else "." for cell in row]) + "\n"
+
+        return map_to_str
+
+    def count_neighbours(self, x, y) -> int:
+        """
+        Counts the number of living cells around a cell.
+        :param x: index of the row (one list in the map)
+        :param y: index of the column (whole map)
+        :return: Number of living cells around the cell.
+        """
+        neighbours = 0
+
+        if len(self.map) == 1:
+            if len(self.map[y]) == 1:
+                return neighbours
+
+        try:
+            self.map[y][x]
+        except IndexError:
+            raise IndexError("Invalid coordinates.") from IndexError
+
+        if y == 0:
+            if len(self.map[y]) == 1:
+                neighbouring_cells = [self.map[y][x]] + [self.map[y + 1][x]]
+            elif x == 0:
+                neighbouring_cells = self.map[y][x + 1 : x + 2] + self.map[y + 1][x : x + 2]
+            elif x != len(self.map[y]) - 1:
+                neighbouring_cells = [self.map[y][x - 1], self.map[y][x + 1]] + self.map[y + 1][x - 1 : x + 2]
+            else:
+                neighbouring_cells = self.map[y][x - 1 : x] + self.map[y + 1][x - 1 : x + 1]
+        elif y != len(self.map) - 1:
+            if x == 0:
+                neighbouring_cells = [self.map[y][x + 1]] + self.map[y - 1][x : x + 2] + self.map[y + 1][x : x + 2]
+            elif x != len(self.map[y]) - 1:
+                neighbouring_cells = (
+                    [self.map[y][x - 1], self.map[y][x + 1]]
+                    + self.map[y - 1][x - 1 : x + 2]
+                    + self.map[y + 1][x - 1 : x + 2]
+                )
+            else:
+                neighbouring_cells = (
+                    self.map[y - 1][x - 1 : x + 1] + self.map[y][x - 1 : x] + self.map[y + 1][x - 1 : x + 1]
+                )
+        else:
+            if x == 0:
+                neighbouring_cells = [self.map[y][x + 1]] + self.map[y - 1][x : x + 2]
+            elif x != len(self.map[y]) - 1:
+                neighbouring_cells = [self.map[y][x - 1], self.map[y][x + 1]] + self.map[y - 1][x - 1 : x + 2]
+            else:
+                neighbouring_cells = self.map[y][x - 1 : x] + self.map[y - 1][x - 1 : x + 1]
+
+        for cell in neighbouring_cells:
+            if cell:
+                neighbours += 1
+            else:
+                continue
+
+        return neighbours
+
+    def next_generation(self):
+        """
+        Calculates the next generation of the map.
+        """
+        pass
+
+
+class Map(AbstractMap):
+    """
+    Class for the map with range = 1 and configurable birth and survival conditions.
+    """
+
+    birth_condition = (list[int],)
+    survival_condition = (list[int],)
```

Comment:
I'm not sure, but here you are defining it as a touple, but later just using it as a list (`self.birth_condition = [3]`)